ModTek/ModTek.cs

```diff
@@ -780,10 +788,19 @@ private static IEnumerator<ProgressReport> InitModsLoop()
                     LogException($"Error: Caught exception while parsing {MOD_JSON_NAME} at path {modDefPath}", e);
                     continue;
                 }
+                
+                BetterLogHandler.Shared.SetupModLog(modDef);
 
                 if (!modDef.ShouldTryLoad(ModDefs.Keys.ToList(), out var reason))
                 {
-                    Log($"Not loading {modDef.Name} because {reason}");
+                    // TODO: double logging could be avoided by attaching the mods fileAppender temporarily to the ModTek HBS logger
+                    // and then just log to the ModTek HBS logger
+                    // that would require to have an ModTek HBS Logger in the first place though
+                    // and one would need to track the file appender of a mod
+                    // and add and remove it depending if we are in the context of a mod again
+                    var text = $"Not loading {modDef.Name} because {reason}";
+                    Log(text);
+                    modDef.Logger.Log($"ModTek: {text}");
```

Comment:
I really would like ModTek to log all mod-related messages also to the mods logger to it appears in the log.txt if the log file is enabled. but as written in the TODO, the simple variant is just doubling the logs which might be not nice but at least would work.

Reply:
Are there other places where doubling the logger is desirable? We could add an optional parameter to ModTek.Logger.Log to take ILog or whatever.

Reply:
When merges start to happen and are successful or fail, when the methods of the mod are called (start,  finish), if dependency resolution succeeded or failed